common/middleware/collection/set-pagination.js

```diff
@@ -0,0 +1,53 @@
+const i18n = require('../../../config/i18n')
+const urlHelpers = require('../../helpers/url')
+
+function setPagination(route) {
+  return function handlePagination(req, res, next) {
+    const totalPages = req.results?.meta?.pagination?.total_pages
+
+    if (totalPages > 1) {
+      const { page = 1 } = req.query
+      const nextPageNumber = parseInt(page) + 1
+      const previousPageNumber = parseInt(page) - 1
+      const previousPage = {
+        href: urlHelpers.compileFromRoute(
+          route,
+          req,
+          {},
+          {
+            page: `${previousPageNumber}`,
+          }
+        ),
+        text: i18n.t('pagination.previous_page'),
+        label: i18n.t('pagination.page_of', {
+          current: previousPageNumber,
+          total: totalPages,
+        }),
+      }
+      const nextPage = {
+        href: urlHelpers.compileFromRoute(
+          route,
+          req,
+          {},
+          {
+            page: `${nextPageNumber}`,
+          }
+        ),
+        text: i18n.t('pagination.next_page'),
+        label: i18n.t('pagination.page_of', {
+          current: nextPageNumber,
+          total: totalPages,
+        }),
+      }
+
+      req.pagination = {
```

Comment:
It would be good to get tests for this result, both within and outside of the page number boundaries.

Reply:
Yep, as mentioned on slack, this was the last module I was finishing tests for. But I wanted to get feedback on the structure :)